tunnel/main.go

```diff
@@ -46,4 +55,71 @@ func WireGuardGenerateKeypair(publicKey *byte, privateKey *byte) {
 	curve25519.ScalarBaseMult(publicKeyArray, privateKeyArray)
 }
 
+func findPhysicalDefaultRoute() (winipcfg.LUID, net.IP, error) {
+	r, err := winipcfg.GetIPForwardTable2(windows.AF_INET)
+	if err != nil {
+		return 0, nil, err
+	}
+	lowestMetric := ^uint32(0)
+	var nextHop net.IP
+	var luid winipcfg.LUID
+	for i := range r {
+		if r[i].DestinationPrefix.PrefixLength != 0 {
+			continue
+		}
+		ifrow, err := r[i].InterfaceLUID.Interface()
+		if err != nil || ifrow.OperStatus != winipcfg.IfOperStatusUp || ifrow.MediaType == winipcfg.NdisMediumIP {
+			continue
+		}
+		if r[i].Metric < lowestMetric {
+			lowestMetric = r[i].Metric
+			nextHop = r[i].NextHop.IP()
+			luid = r[i].InterfaceLUID
+		}
+	}
+	if len(nextHop) == 0 {
+		return 0, nil, errors.New("Unable to find default route")
+	}
+	return luid, nextHop, nil
+}
+
+//export TestConnectivity
+func TestConnectivity() int32 {
+	const (
```

Comment:
Will probably parameterize this later.
Fun fact: Moz has their own captive portal detection URL for Firefox: http://detectportal.firefox.com/success.txt